genesyscloud/workforcemanagement_businessunits/resource_genesyscloud_workforcemanagement_businessunits.go

```diff
@@ -0,0 +1,128 @@
+func getAllAuthWorkforcemanagementBusinessunits(ctx context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	proxy := newWorkforcemanagementBusinessunitsProxy(clientConfig)
+	resources := make(resourceExporter.ResourceIDMetaMap)
+
+	businessUnitResponses, resp, err := proxy.getAllWorkforcemanagementBusinessunits(ctx)
+	if err != nil {
+		return nil, util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to get workforcemanagement businessunits: %v", err), resp)
+	}
+
+	for _, businessUnitResponse := range *businessUnitResponses {
+		resources[*businessUnitResponse.Id] = &resourceExporter.ResourceMeta{BlockLabel: *businessUnitResponse.Name}
+	}
+
+	return resources, nil
+}
+
+// createWorkforcemanagementBusinessUnit is used by the workforcemanagement_businessunits resource to create Genesys cloud workforcemanagement businessunits
+func createWorkforcemanagementBusinessUnit(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getWorkforcemanagementBusinessunitsProxy(sdkConfig)
+
+	createBusinessUnitRequest := getCreateWorkforcemanagementBusinessUnitRequestFromResourceData(d)
+
+	log.Printf("Creating workforcemanagement businessunits %s", *createBusinessUnitRequest.Name)
+	businessUnitResponse, resp, err := proxy.createWorkforcemanagementBusinessunits(ctx, &createBusinessUnitRequest)
+	if err != nil {
+		return util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to create workforcemanagement businessunits: %s", err), resp)
+	}
+
+	d.SetId(*businessUnitResponse.Id)
+	log.Printf("Created workforcemanagement businessunits %s", *businessUnitResponse.Id)
+	return readWorkforcemanagementBusinessunits(ctx, d, meta)
+}
+
+// readWorkforcemanagementBusinessunits is used by the workforcemanagement_businessunits resource to read a workforcemanagement businessunits from genesys cloud
+func readWorkforcemanagementBusinessunits(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getWorkforcemanagementBusinessunitsProxy(sdkConfig)
+	cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceWorkforcemanagementBusinessunits(), constants.ConsistencyChecks(), resourceName)
+
+	log.Printf("Reading workforcemanagement businessunits %s", d.Id())
+
+	return util.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		businessUnitResponse, resp, getErr := proxy.getWorkforcemanagementBusinessunitsById(ctx, d.Id())
+		if getErr != nil {
+			if util.IsStatus404(resp) {
+				return util.RetryableErrorWithRetryAfter(ctx, util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("Failed to read workforcemanagement businessunits %s: %s", d.Id(), getErr), resp), resp)
```

Comment:
Am I reading this wrong, or why are we saying that a 404 is a retryable error?

Reply:
updated in next commit